### PR TITLE
Improve default compiler story for wasmer cli

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
This PR needs the following to be ready to ship:

- [x] Verified make commands are still fine
- [x] Verified this doesn't do anything bad in CI

This PR adds:
- compile_error if no backend is specified in wasmer.rs
- readds clif as the default
- improves the error message saying "specified backend not found" to include the name of the backend that it couldn't find

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

File: src/bin/wasmer.rs [hunk 1]
```diff
@@ -40,6 +40,13 @@ use wasmer_runtime_core::{
 #[cfg(feature = "wasi")]
 use wasmer_wasi;
 
+#[cfg(all(
+    not(feature = "backend-cranelift"),
+    not(feature = "backend-llvm"),
+    not(feature = "backend-singlepass")
+))]
```
Comment (syrusakbary) [inline src/bin/wasmer.rs]: I think this is a bit easier to read:

```suggestion
#[cfg(not(any(
    feature = "backend-cranelift",
    feature = "backend-llvm",
    feature = "backend-singlepass"
)))]
```

File: src/bin/wasmer.rs [hunk 2]
```diff
@@ -47,6 +47,13 @@ use std::{cell::RefCell, io::Write, rc::Rc};
 #[cfg(feature = "backend-llvm")]
 use wasmer_runtime_core::backend::BackendCompilerConfig;
 
+#[cfg(not(any(
+    feature = "backend-cranelift",
+    feature = "backend-llvm",
+    feature = "backend-singlepass"
+)))]
+compile_error!("Please enable at least one of the compiler backends");
```
Comment (nlewycky) [inline src/bin/wasmer.rs]: Optional: s/at least one/one or more/

**Pull request merged**